crates/nu-protocol/src/ast/block.rs

```diff
@@ -55,6 +55,17 @@ impl Block {
             recursive: None,
         }
     }
+
+    pub fn new_with_capacity(capacity: usize) -> Self {
+        Self {
+            signature: Box::new(Signature::new("")),
+            pipelines: Vec::with_capacity(capacity),
```

Comment:
This sounds elegant! I am not sure if the `From<Iterator<Item = Pipeline>>` implementation below with its `FromIterator`/`.collect()` internally already achieves that as `.map()` in the old impl should preserve the size hint. (but specialization is still a secret sorcery to me)

Reply:
this one was more of a personal preference because I didn't like seeing the `.into()` taking up so much space in the flamegraph 😅